packages/components/README.md

```diff
@@ -37,35 +41,71 @@ Installation is simple. All you need to do is add `<courier>` components to your
+
+| Key        | Type              | Description                                                                                                           |
+| ---------- | ----------------- | --------------------------------------------------------------------------------------------------------------------- |
+| clientKey  | `string`          | Key associated with your account. Found on https://app.courier.com/integrations/courier                               |
+| userId     | `string`          | The current user logged into your app. Associated with Courier's `recipientId`                                        |
+| initOnLoad | `boolean`         | If you don't want Courier to try and render the components right away, you can pass this flag to defer initialization |
+| components | `ComponentConfig` | Map of configuration for each component (`toast` and `inbox`) on the page                                             |
+
+> The components will not render unless we have both the `userId` and `clientKey`
 
-To interact with the components, we have exposed an sdk on `window.courierSdk`. Since this sdk initializes asynchronous, you can create and push initialization functions onto an array named `window.courierAsyncInit`.
+## [Asynchronous Initialization](#async-init)
+
+To interact with Courier and its components, we have exposed an sdk on `window.courierSdk`. Since this sdk initializes asynchronous, you can create and push initialization functions onto an array named `window.courierAsyncInit`. When the sdk is ready we will call all the functions on this array.
 
 Example:
 
 ```html
 <script type="text/javascript">
-  window.courierConfig = {
-    clientKey: "{{CLIENT_KEY}}",
-    userId: "{{USER_ID}}",
-  };
-
   window.courierAsyncInit = window.courierAsyncInit || [];
   window.courierAsyncInit.push(() => {
-    window.courierSdk.toast({
-      title: "Hello",
-      body: "World",
-    });
-
-    window.courierSdk.transport.intercept((message) => {
-      const iWantToHideThisMessage = true;
-      if (iWantToHideThisMessage) {
-        return;
-      }
-
-      // you can also mutate this message before displaying to a user
-      return message;
-    });
+    console.log("Courier is Ready!");
+  });
+</script>
+```
+
+## [Courier SDK](#courier-sdk)
+
+After Courier has initialized, the object `window.courierSdk` is ready.
+
+- window.courierSdk.init(config);
+
+Use this function to initialize the rendering of the Courier components if you were not able to initialize synchronously.
+
+- window.courierSdk.on((action, cb));
+
+To listen for actions that happen inside Courier's SDK.
+
+## [SDK Actions](#sdk-actions)
+
+- `toast/init`
+
+```html
+<script>
+  window.courierAsyncInit.push(() => {
+    window.courierSdk.on("toast/init", () => {
+      window.courierSdk.toast({
+        title: "Hello",
+        body: "World",
+      });
+    };
+  });
+</script>
+```
+
+- `inbox/init`
+
+```html
+<script>
+  window.courierAsyncInit.push(() => {
```

Comment:
A function is being pushed on to a stack of? Seems kind of arbitrary. As a developer I wouldn't know at what part of the lifecycle this is being called. In example, when the script loads, or after I have called the `init` function